### PR TITLE
txn: stop more statement executions if pessimistic transaction ttl manager timed out

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve?

Problem Summary:

Allow only `commit` or `rollback` if pessimistic transaction TTL manager has timed out

### What is changed and how it works?

What's Changed:

1. fix the problem that `twoPhaseCommitter`.`isPessimistic` field has incorrect value as it's not intialized before `LockKeys`
2. Set timed out flag in `TxnCtx` and check this flag before executing another statement, for example in 
`RC` mode, users could not do more read since the lock cache has expired

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)
1. start transaction 
2. add pessimistic locks
3. let ttl manager time out
4. try to do more executions 

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### CODE REVIEW

File: errno/errcode.go [hunk 1]
```diff
@@ -1069,6 +1069,7 @@ const (
 	ErrCannotCancelDDLJob             = 8226
 	ErrSequenceUnsupportedTableOption = 8227
 	ErrColumnTypeUnsupportedNextValue = 8228
+	ErrLockExpire                     = 8229
```
Comment (jackysp) [inline errno/errcode.go]: We may update the docs for it.

**Comment by coocood** [comment]
LGTM

**Pull request merged**